src/jit_channel/msgs.rs

```diff
@@ -0,0 +1,263 @@
+use std::convert::TryFrom;
+
+use bitcoin::hashes::hmac::{Hmac, HmacEngine};
+use bitcoin::hashes::sha256::Hash as Sha256;
+use bitcoin::hashes::{Hash, HashEngine};
+use serde::{Deserialize, Serialize};
+
+use crate::transport::msgs::{LSPSMessage, RequestId, ResponseError};
+use crate::utils;
+
+pub(crate) const LSPS2_GET_VERSIONS_METHOD_NAME: &str = "lsps2.get_versions";
+pub(crate) const LSPS2_GET_INFO_METHOD_NAME: &str = "lsps2.get_info";
+pub(crate) const LSPS2_BUY_METHOD_NAME: &str = "lsps2.buy";
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize, Default)]
+#[serde(default)]
+pub struct GetVersionsRequest {}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct GetVersionsResponse {
+	pub versions: Vec<u16>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct GetInfoRequest {
+	pub version: u16,
+	pub token: Option<String>,
+}
+
+/// Fees and parameters for a JIT Channel.
+///
+/// The client will pay max(min_fee_msat, proportional*(payment_size_msat/1_000_000)).
+pub struct RawOpeningFeeParams {
+	/// The minimum fee required for the channel open.
+	pub min_fee_msat: u64,
+	/// A fee proportional to the size of the initial payment.
+	pub proportional: u32,
+	/// An ISO8601 formatted date for which these params are valid.
+	pub valid_until: String,
+	/// number of blocks that the LSP promises it will keep the channel alive without closing, after confirmation.
+	pub min_lifetime: u32,
+	/// Maximum number of blocks that the client is allowed to set its to_self_delay parameter.
+	pub max_client_to_self_delay: u32,
+}
+
+impl RawOpeningFeeParams {
+	pub(crate) fn into_opening_fee_params(self, promise_secret: &[u8; 32]) -> OpeningFeeParams {
+		let mut hmac = HmacEngine::<Sha256>::new(promise_secret);
+		hmac.input(&self.min_fee_msat.to_be_bytes());
+		hmac.input(&self.proportional.to_be_bytes());
+		hmac.input(self.valid_until.as_bytes());
+		hmac.input(&self.min_lifetime.to_be_bytes());
+		hmac.input(&self.max_client_to_self_delay.to_be_bytes());
+		let promise_bytes = Hmac::from_engine(hmac).into_inner();
+		let promise = utils::hex_str(&promise_bytes[..]);
+		OpeningFeeParams {
+			min_fee_msat: self.min_fee_msat,
+			proportional: self.proportional,
+			valid_until: self.valid_until.clone(),
+			min_lifetime: self.min_lifetime,
+			max_client_to_self_delay: self.max_client_to_self_delay,
+			promise,
+		}
+	}
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+/// Fees and parameters for a JIT Channel with promise.
```

Comment:
yeah, actually I think OpeningFeeParams (with promise) should not be exposed at all.  We should probably have a From<OpeningFeeParams> for RawOpeningFeeParams where we strip out the promise before surfacing it to the end-user to select a fee.

Reply:
I guess the challenge of doing that is that we actually need to send back the fee param with promise to the LSP when we issue the BuyRequest.  This means we would have to keep around the menu in the InboundJitChannelState and match it to the one the end-user picked as part of `opening_fee_params_selected`.

 Not a huge deal.. think it's worth it to avoid having to document and expose the `promise` concept?